### PR TITLE
Closes Sub-Task CPPL-128

### PR DESCRIPTION
**Pull request by VasylDobrovolskyi**
Implemented GetLists Handler

### CODE REVIEW

**Comment by iodream** [comment]
Seems to be good but what about the URI on the handler factory?

File: src/Server/Handlers/GetListsHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	{
+		std::vector<List> lists = vector.value();
+		QJsonArray jsonarray;
+		for (int i = 0; i < int(lists.size()); i++)
+		{
+			QJsonObject json;
+			json["id"] = lists[i].list_id;
+			json["ownerId"] = lists[i].owner_id;
+			json["name"] = QString::fromStdString(lists[i].name);
+			auto list_state = m_facade->GetListStateById(lists[i].state_id);
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/GetListsHandler.cpp:35]: Db request within Formatter roughly violates srp. Please provide another solution.(Consider https://github.com/iodream/Spend-Note/pull/25/files)

File: src/Server/Handlers/GetListsHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	QJsonObject finaljson;
+	if (vector.has_value())
+	{
+		std::vector<List> lists = vector.value();
+		QJsonArray jsonarray;
+		for (int i = 0; i < int(lists.size()); i++)
+		{
+			QJsonObject json;
+			json["id"] = lists[i].list_id;
+			json["ownerId"] = lists[i].owner_id;
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/GetListsHandler.cpp:33]: Use lowercase notation for json field names.

File: src/Server/Handlers/GetListsHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{
+	QJsonObject finaljson;
+	if (vector.has_value())
+	{
+		std::vector<List> lists = vector.value();
+		QJsonArray jsonarray;
+		for (int i = 0; i < int(lists.size()); i++)
+		{
+			QJsonObject json;
+			json["id"] = lists[i].list_id;
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/GetListsHandler.cpp:32]: Currently we are using string for ids in json, so replace all such occurrences with std::to_string(id_name).c_str().

File: src/Server/Handlers/GetListsHandler.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#include "../libdal/Facade/IDbFacade.h"
+#include "../libdal/DTOs/List.h"
+
+using IdType= long long int;
+
+class GetListsHandler : public AuthorizedHandler
+{
+	QJsonDocument Format(const std::optional<std::vector<List>>& vector);
+	GetListsHandler();
+    virtual ~GetListsHandler() override {}
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/GetListsHandler.h:16]: Indentation issue

File: src/Server/Handlers/GetListsHandler.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    virtual Net::Response AuthHandle(const Net::Request& request) override;
+	std::unique_ptr<IDbFacade> m_facade;
```
Comment (Denys-Bykov-0) [inline src/Server/Handlers/GetListsHandler.h:18]: Should be removed, after migration to the new ICommandHandler

**Review by Denys-Bykov-0 — COMMENTED** [review]
Requires migration to the new ICommandHandler from https://github.com/iodream/Spend-Note/pull/25.

**Review by Denys-Bykov-0 — APPROVED** [review]
Should be ok, but fix indentation before merging. (You can use whitespace display in your IDE for this purpose)

**Review by iodream — APPROVED** [review]

**Pull request merged**